harness/dotnet/Temporalio.Features.Harness/Runner.cs

```diff
@@ -3,37 +3,64 @@ namespace Temporalio.Features.Harness;
 using Temporalio.Client;
 using Temporalio.Exceptions;
 using Temporalio.Worker;
-using Temporalio.Workflows;
 
 /// <summary>
 /// Runner for running features.
 /// </summary>
 public class Runner
 {
     private bool? maybeUpdateSupported;
+    private ITemporalClient? client;
+    private WorkerState? workerState;
+    private CancellationToken? runCancelToken;
+
+    class WorkerState
+    {
+        private readonly CancellationTokenSource workerStopper;
+        private readonly Task workerTask;
+
+        public WorkerState(
+            CancellationTokenSource workerStopper,
+            TemporalWorker worker
+        )
+        {
+            this.workerStopper = workerStopper;
+            workerTask = worker.ExecuteAsync(workerStopper.Token);
```

Comment:
```suggestion
            workerTask = Task.Run(
                async () =>
                {
                    try
                    {
                        await worker.ExecuteAsync(workerStopper.Token);
                    }
                    catch (Exception e)
                    {
                        // Log e
                        throw;
                    }
                });
```

The existing code is benefitted by the way we wrote the internals of this call that it's not an `async`, but by default in .NET tasks don't start just because you invoke a method that returns a task, so `Task.Run` is best.

Also, technically the problem with the split start/stop approach (and why we avoided it in the .NET worker API) is because you can't see fatal worker exceptions. So that's what the catch here is for. But if we aren't concerned with that, no prob.

Reply:
It's not a matter of not being concerned with it - it's that if I need to stop a worker and start a new one at any time, the `RunUntil` approach just doesn't really fit.

In fact it would be really nice to have a way to stop the worker that doesn't force an exception to be thrown. That feels like a bit of an artificial burden.

Reply:
This seems to work jut fine without Task.Run too - I thought .NET always eagerly Ran tasks? In any case I put it inside Task.Run but w/o the try/catch since that would log every time it's cancelled and would need the special-case for that too, and the task gets awaited on anyway, so any thrown exception will bubble up.

Reply:
> I thought .NET always eagerly Ran tasks?
No, unfortunately not (same with Python), but if you write a method that returns a `Task` instead of an `async` method that returns a task, that runs eagerly. That's why this worked before, because that's what we did internally.

> but w/o the try/catch since that would log every time it's cancelled and would need the special-case for that too, and the task gets awaited on anyway, so any thrown exception will bubble up.

But the exception isn't bubbled up anywhere until `StopAndWait`. But I am fine with that if we are. Also, swallowing a cancel here is easier than swallowing it in your other two places I think.